iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheck.java

```diff
@@ -20,91 +20,100 @@
 
 @Rule(key = "S6865")
-public class AutomountServiceAccountTokenCheck extends AbstractResourceManagementCheck<ServiceAccount> {
-  private static final String MESSAGE = "Set \"automountServiceAccountToken\" to false for this specification of kind %s.";
-  private static final String KEY = "automountServiceAccountToken";
+public class AutomountServiceAccountTokenCheck extends AbstractGlobalResourceCheck {
+  private static final String MESSAGE_BIND_ACCOUNT_RESOURCE = "Bind this Service Account to RBAC or disable \"automountServiceAccountToken\".";
+  private static final String MESSAGE_BIND_ACCOUNT_NAME = "Bind this resource's automounted service account to RBAC or disable automounting.";
   private static final String KIND_POD = "Pod";
+  private static final String SERVICE_ACCOUNT_NAME = "serviceAccountName";
   private static final List<String> KIND_WITH_TEMPLATE = List.of("DaemonSet", "Deployment", "Job", "ReplicaSet", "ReplicationController", "StatefulSet", "CronJob");
+  private static final Predicate<YamlTree> IS_FALSE = tree -> "false".equals(TextUtils.getValue(tree).orElse(""));
 
   @Override
   void registerObjectCheck() {
-    register(KIND_POD, document -> checkAndReport(document, String.format(MESSAGE, KIND_POD)));
+    register(KIND_POD, document -> checkResource(document, document.block("spec")));
+    register(KIND_WITH_TEMPLATE, document -> checkResource(document, document.block("spec").block("template").block("spec")));
+  }
 
-    for (String kind : KIND_WITH_TEMPLATE) {
-      register(kind, (BlockObject document) -> checkAndReport(document.block("spec").block("template"), String.format(MESSAGE, kind)));
+  private void checkResource(BlockObject document, BlockObject spec) {
+    if (isAutomountDisabled(spec) || isContainersAbsent(spec)) {
+      return;
     }
+    var namespace = CheckUtils.retrieveNamespace(document);
+    var accountName = CheckUtils.retrieveAttributeAsString(spec, SERVICE_ACCOUNT_NAME);
+    checkAccountSecurity(document, spec, namespace, accountName);
   }
 
-  @Override
-  Class<ServiceAccount> getGlobalResourceType() {
-    return ServiceAccount.class;
+  private static boolean isAutomountDisabled(BlockObject spec) {
+    return spec.attribute("automountServiceAccountToken").isValue(IS_FALSE);
   }
 
-  private void checkAndReport(BlockObject blockObject, String message) {
-    var specAsBlockObject = blockObject.block("spec");
-    var specAsAttributeObject = blockObject.attribute("spec");
-    if (specAsAttributeObject.tree != null && isContainersPresentInSpecBlock(specAsBlockObject)) {
-      var tokenAttribute = specAsBlockObject.attribute(KEY);
-      if (!tokenAttribute.isAbsent()) {
-        tokenAttribute.reportIfValue(TreePredicates.isSet().negate(), message);
-        tokenAttribute.reportIfValue(isTrue(), message);
-        return;
-      }
+  private static boolean isContainersAbsent(BlockObject spec) {
+    return spec.attribute("containers").tree == null;
```

Comment:
We have a method `isAbsent` on `AttributeObject`, will it work here?

Reply:
Ah indeed, good catch! I kept this method from the previous implementation, I guess there was not `isAbsent(...)` at that time.